src/main/clojure/clara/rules/engine.cljc

```diff
@@ -2002,6 +2008,29 @@
                  get-alphas-fn
                  []))
 
+(defn with-listener
+  [session listener]
+  (let [{:keys [listeners] :as components} (components session)]
+    (assemble (assoc components
+                     :listeners
+                     (conj listeners
+                           listener)))))
+
+(defn remove-listeners
+  [session pred]
+  (let [{:keys [listeners] :as components} (components session)
+        matching-listeners (filterv pred listeners)]
```

Comment:
I think using `some` here would make more sense. No reason to build a vector of all the matches that aren't used later.